test/CumulativeMerkle.test.js

```diff
@@ -320,10 +335,24 @@ describe('Cumulative Merkle Distribution', function () {
               const claimAccount = cumulativeProofAccounts[index]
               const claimAmount = ethers.BigNumber.from(cumDist.claims[claimAccount].amount)
               const claimProof = cumDist.claims[claimAccount].proof
-
-              await merkleDist.claim(claimAccount, claimAmount, cumulativeMerkleRoot, claimProof)
-              const balance = await token.balanceOf(claimAccount)
-              expect(claimAmount).to.equal(balance)
+              const claimBeneficiary = cumDist.claims[claimAccount].beneficiary
+              
+              // add up all rewards from previous distribution
+              let prevReward = {}
+              proofAccounts.forEach((account, _) => {
+                if (dist.claims[account].beneficiary === claimBeneficiary) {
+                  prevReward[account] = dist.claims[account].amount
+                }
+              })
+
+              // update reward for current distribution
+              prevReward[claimAccount] = claimAmount
+              const reducer = (accumulator, curr) => parseInt(accumulator, 10) + curr;
```

Comment:
If we run Hardhat tests with a greater number of stakes (like this new example: https://github.com/theref/merkle-distribution/pull/4), this test will fail. It seems like the calculation of `balance` or `totalReward` is not running with the needed precision:

```
  Stack trace: AssertionError: Expected "958346389" to be equal 969346389
```

Could it be possible that the problem is the use of Integer types instead of BigNumber in this reducer function?

Reply:
it's to do with state changes in the tests. For example, if the beneficiary for stakeholder changes between the first distribution and the second distribution, within the test you need to calculate the difference between `amount` from `cumDist` and `dist`, which means tracking the first distribution.

I deliberately simplified `stake_list_20` in order to avoid this. The tests need to be much more sophisticated to handle a complicated cumulative distribution.

Reply:
To be clear, I think that work should be done, but I'm happy with the tests we've got for now

Reply:
Got it! Maybe we should move this to a new issue to deal with it later on.